### DISCUSSION TITLE
Two factor auth failing, even on setup

### DISCUSSION THREAD

**Discussion by faunam** [community/community#22013] [New to GitHub]
<p>My two factor auth codes suddenly failed early last week, i didn’t have my recovery codes so I started the account recovery process. Now I’m trying to set up two factor auth again, and it is still failing, even on setup (when I scan the QR code on an app and enter the codes into github, it says “Two-factor code verification failed. Please try again.”)<br>
I don’t know what to do at this point, this is my work account and I need it working again ASAP, it’s really slowing me down.<br>
I’ve temporarily used SMS, which works, but I will be travelling soon and I’m not sure if it will work then, so would prefer to get this working with an app.</p>
<p>Thanks,<br>
Fauna</p>

**Comment by faunam** [comment, marked as answer]
<p>Resolved! according to <a href="https://github.com/google/google-authenticator-android/issues/56" rel="noopener nofollow ugc">this thread</a>, in my authenticator app, under ‘settings’ I tapped ‘Time Correction for Codes’ → ‘Sync now’ and that worked like a charm</p>

**Reply by creativecode01** [reply]
wow you are my savior. thanks

**Reply by MortezaMaghrebi** [reply]
ابتدا در یک اکانت گیت هاب که از قبل لاگین بودین، وارد ستینگ بشین و در قسمت password and authentication، روی دکمه Edit قسمت Authenticator App بزنید، بعد نکته مهم اینه که تو تنظیمات گوشیتون برید و ساعت گوشی رو با اینترنت sync کنید، چون برنامه Authenticator از طریق زمان یک کد میسازه و باید حتما ساعت گوشیتون با ساعت گیت هاب سینک باشه، برای این کار set time automatically رو در تنظیمات date time گوشیتون غیر فعال و مجددا فعال کنین. حالا برنامه Microsoft Authenticator رو باز کنین و QR Code رو اسکن کنین، و بعد عدد 6 رقمی رو بدون فاصله در قسمت Verify the code from the app تایپ کنین و اگه درست بود حتما Save رو بزنین. از این به بعد میتونین لاگین کنین.
یه کار دیگه که کردم ولی فکر میکنم ربطی نداره به این مشکل اینه که تو تنظیمات گیت هاب در قسمت public profile، تیک Display current local time رو زدم و Time Zone رو روی Tehran تنظیم کردم.

**Comment by canuckjacq** [comment]
<p>:wave: Welcome!</p>
<p>You haven’t mentioned which app you are using, but one of the reasons we sometimes see failures to set up TOTP is if the device you’re using has the time set manually, and is out of sync with the server. That can cause it to create incorrect codes.</p>

**Reply by lukecapizano** [reply]
Thank you! I had my Windows PC set to automatically resolve the time. But looking at it vs my laptop, I noticed it was slightly ahead. Re-syncing my PC's local time fixed my issue!

**Comment by canuckjacq** [comment]
<p>Oh! That’s great! Thanks for updating :slight_smile:</p>

**Comment by faunam** [comment]
<p>LOL  yeah initially i was using google authenticator. though, i installed lastpass authenticator just to see if a different app would change things, and it still failed. it’s surprising to me that I would need to time sync immediately. after installing</p>

**Comment by canuckjacq** [comment]
<p>It’s the only real problem I’ve heard of these apps having, and that’s a good thing!</p>

**Comment by dpapkin** [comment]
<p>I am using Microsoft authenticator and it is failing. There is no Time Sync option under Settings, Both my laptop and my samsung android phone are syncing time automatically</p>

**Reply by XinLiTCD** [reply]
yes,no Time Sync Option. So did you fix the issue?

**Reply by JianWillWang** [reply]
the same, no Time sync option by microsoft authenticator

**Reply by bushnerd** [reply]
the same, no Time sync option by microsoft authenticator.
And i set the date and time to automatic on ios, it does not work.

**Comment by rfscarlat** [comment]
I use Google Authenticator, which is in sync with Google time servers, and it still fails. I always have to use the recovery codes. The solution doesn't work, as the app reports that it is in sync.

**Reply by SarkisKhachatryan** [reply]
Same here, do you find any solution to this?

**Reply by dipanshparmar** [reply]
I had to restart the app after registering the github account in Google Authenticator. After restarting the app, the code I got worked fine.

**Comment by rLry** [comment]
There is also no Time Sync option under Settings in my keepassXC

**Comment by littlehesus** [comment]
Experienced the same problem randomly , changed phones , from android to ios and everything seemed to be working just fine with the Authenticator app , yesterday and today , couldnt accept my code , everything seemed to be on sync , so I opened my old phone and used that code and worked like a charm. not sure why though and passed the 2FA on the privacyIDEA Authenticator if I will have better luck. For now though I disabled it.

**Comment by muneib** [comment]
I use Microsoft Authenticator and had the same issue. To solve this, I went to my mobile setting and set the date and time to automatic.

**Reply by Mike-Schvedov** [reply]
You are a life saver.

**Reply by IdelsonMendes** [reply]
thx bro

**Reply by HarrisonFok** [reply]
My date and time are defaulted to Automatic, but it's still failing for some reasons
It was working fine earlier in the year

**Reply by luizfino13** [reply]
It worked for me 2. 
God bless u, bro.

**Reply by Gerardo-CLL** [reply]
NICE!

**Reply by RajKousik** [reply]
It worked like a charm! Thanks

**Reply by talhaanwarch** [reply]
worked

**Reply by jtrollkarl** [reply]
for anyone else frustrated/freaking out - this solution also worked for me. using android and M$ authenticator

**Comment by iusztin** [comment]
I use Authy. Same issue. No Time Sync option.

**Comment by israebe** [comment]
I use Microsoft Authenticator and had the same issue. To solve this, I went to my mobile setting and set the date and time to automatic.

**Reply by zandiarash** [reply]
Yes, That works for me, Thank you a lot.
ایرانیای عزیز - چیز شکن هم باید فعال کنین.

**Comment by looeee** [comment]
I'm using Microsoft Authenticator. 
I've tried it on two different phone (a Redmi Note 10 Pro and a Galaxy s21FE). Both have date and time set to automatic. 

Neither of them work. I get the message "Two-factor authentication failed."

I've tried signing into Github using 2FA on both my laptop and my phone. Both of which have the time and date set to the same as the MS Authenticator app. 

Is there anything else I can try? I use MS Authenticator for quite a few things, and only Github has this issue.

**Comment by Tridu33** [comment]
the same problem here

**Comment by sclcoder** [comment]
I had the same problem, but the difference was that I had multiple devices set up. Only the last device's authorization code is correct. For example, I first added the configuration for Microsoft's Authoenticator, and then added the configuration in the Chrome authentication plug-in. Only the configuration in Chrome worked. Finally, I added the authorized ID in chrome to Microsoft Authoenticator, so that the authorization codes of the two terminals are consistent.

**Comment by SNWCreations** [comment]
Same problem. I'm using Microsoft authenticator and I've checked the time settings of my phone and computer, they are all correct.
This issue just happened to me recently, nothing went wrong before.
What should I do next?

**Reply by SNWCreations** [reply]
My phone now switched to Google Authenticator, and still doesn't work.

**Comment by jaseemtan** [comment]
~~Not working with Microsoft Authenticator on iOS.~~
Got it working. I removed the setup and added again. And this time it worked.

**Reply by derrondp** [reply]
thanks. this worked

**Comment by kulapati** [comment]
The same problem occurred. There were no issues before, even though I installed Microsoft and Google authenticator apps on a new Android phone a month ago. But it didn't work yesterday, I adjusted the time but it didn't help. However, when I tried the old phone's Microsoft authenticator app, it surprisingly worked. Do I have to use the old phone for authentication? Is this reasonable?

**Reply by kulapati** [reply]
Does it mean I can only use the authenticator that was first used? I also tried to authenticate using a browser plugin, but it failed.

**Reply by kulapati** [reply]
I have installed both Microsoft and Google authenticators on two different Android phones, and yesterday I installed the plugin on the Edge browser. Upon testing, I found that each time, the login code displayed is different. Only the one installed earliest in the Microsoft authenticator can login to GitHub, the others fail. Is this normal? I haven't seen any clear indication that it can only be installed on one mobile device.

Initially, I thought it might be a time synchronization issue, but I found that it wasn't the case, as the time is synchronized. Even on the same phone, different authenticators still cannot log in. Perhaps it really can only be installed on one device and one app? What would happen if I delete the authenticator on the old phone?

**Comment by FxBresson** [comment]
As suggested by @muneib, to solve this I had to go to my phone Date & Time settings, set time to manual, set again to automatic, remove the Github entry in my authenticator app. Then I added Github again with the QA code, and it finally worked ! :D

**Reply by klh315** [reply]
I tried this way. Delete github entry, and readd it works well

**Reply by jtgsothis** [reply]
This worked for me too, thanks!
Additionally, I would recommend to register an alternative MFA method (SMS) before removing the one from the app, just in case.

**Comment by SNWCreations** [comment]
I've solved this problem.
I uninstalled the authenticator by mistake before.
After reinstalling it, I've scanned the QR code provided in the security page, but I forget to enter the confirm code. So GitHub don't recognize my authenticator :/
In a word, DO NOT FORGET TO ENTER CONFIRM CODE WHEN RECONFIGURING.
Confirm code is just the code provided by authenticator.

**Reply by gzm8866007** [reply]
Thank you, I solved it using your method.

**Comment by chansonzhang** [comment]
I met the same problem in Microsoft Authenticator, remove and re-add account solved the problem. I guess only one auth app (the last configured one) can be used for github 2FA, so if you add configure the 2fa in 2 apps (for me they are twilio authy and microsoft authenticator), one of them will be invalid.

**Comment by qrp73** [comment]
The same issue, attempt to setup 2FA fails, because it shows "Two-factor code verification failed. Please try again." just on the first attempt after barcode scan with Google Authenticator app.

Is 2FA broken on github?

The only way to login into github now is to use recovery codes. 2FA don't works at all - it reject valid 2FA codes.

Update: fixed. The root of cause - incorrect time zone setting on smartphone. It showed correct local time, but it seems that app using UTC time which is incorrect in that case.

**Comment by am11** [comment]
After scanning the QR code, make sure to input TOTP code in `Verify the code from the app` box. That code shows up in your authenticator app under `GitHub` after the QR scan, and then hit <kbd>Save</kbd> button. It was a silly thing I was missing and I found this UX unhelpful because GitHub keeps showing us (irreversible) `Configured` state from the first time we enabled TOTP way back when, and it doesn't allow us to disconnect-from-old-and-reconnect-to-new-device to reflect that the setup on new device hasn't been completed:
<img width="264" alt="image" src="https://github.com/community/community/assets/3840695/39a1510d-1d75-40a5-ab41-1a59c401cc1c">

**Reply by igor-m** [reply]
Hi, so my question is following:
I have got 3 computers (different locations), and I want to operate my single github account out of them.
I've got a single "Google Authenticator" app on my single mobile phone, and the "Authenticator" extensions in my FireFox browsers installed on all 3 computers.
How should I proceed to have the stuff working on all 3 computers - via my mobile phone and/or via the 3 Firefoxes???

**Reply by am11** [reply]
I also have three devices (two laptops and a phone). A few weeks ago, I was using this method:
* navigate to https://github.com/settings/security on a computer and configure `Authenticatior app` (where it shows the QR code)
* open the navigator app on the phone and  scan QR code
* once successfully scanned, it will start showing TOTP (which expires in a few seconds). Input that code in the box next to `Verify the code from the app` under the QR code on `settings/security` page and hit the <kbd>Save</kbd> button

Each time I was signing in on a new device/browser, I was using the code from authenticator app. Mostly, I use the same browser on each device and my credentials are saved by the browser, so I seldom have to sign in.

Now, I have _also_ installed `GitHub Mobile` app on my phone and registered it on the same `settings/security` page and have selected GitHub Mobile as a `Preferred 2FA method`. Each time I have to sign in (on a new device or, e.g., after resetting the browser to factory settings), it asks me to open the GitHub Modile and input a number shown on the screen. If in case GitHub Mobile is inaccessible, it will fallback to TOTP (Authenticator app), and then to the recovery code option.

I recommend registering both options (Authenticator app and GitHub Mobile) if you can, plus securing the recovery keys list (which can be obtained from the same `settings/security` page). A few times, I didn't had access to my phone where those recovery codes helped.

**Reply by igor-m** [reply]
Yep, I've spent couple of hours to mess with it and indeed this works, it seems:
1. on the PC_1 go to your github account->settings->edit 2FA (after the opening it you get a new QR "seed"), do NOT "save the code for verification" yet, the page must stay opened during the entire exercise
2.  copy the QR code with a snipping tool into a .jpeg file
3.  read the QR with your Google Authenticator (in the smartphone for example), it will tell you the code has been saved, but do NOT "enter and save the code for verification" at that page
4. transfer the jpeg picture with your QR code to the PC_2 and PC_3 (for example). Open your Firefox(es) with the "Authenticator" extension in it, any page opened is ok, and do drop the jpeg picture into their pages (such you see the picture in the FF). Open the FF's Authenticator(s) in the PC_1/2/3 for adding a new account and "scan" the QR code (it "scans" inside the FF's page with the mouse only), it will read the QR and it will add the info to its accounts list.
5. return to the PC_1, look a the current number code in your smartphone (in the GA) and do enter it into the field in the opened page (see step 1.) to verify the QR code and press the "Save" finally..
6. at the end of this exercise all the authentication gadgets you played with shall show the same number code in the given moment in time..

**Comment by mikeblas** [comment]
I'm trying to setup Google Authenticator on an Android Phone because I'm getting the scary "permanent account lockout" message. Authenticator ate the 2-D barcode just fine, but any "code from the app" I enter does work and gives the message "Two-factor code verification failed. Please try again." 

This persists after using the "Sync now" option for time codes in the app. (And that gets "Time already correct" from the phone app, anyway.)

How can I set up 2FA and avoid having my account permanently locked?

**Comment by qrp73** [comment]
For those who has issue with Google Authenticator. Make sure you have proper time zone and proper time on the phone. Often the issue is due to wrong time zone in system settings.

**Comment by pdalfarr** [comment]
New phone here and I encountered the same problem.
On my old phone I have Microsoft Authenticator, which is working fine.
But running two-factor authentication on GitHub using the codes provided by Microsoft Authenticator **from my new phone** simply fails.
Time Zone setting is correct (and identical) on both phone.
I uninstalled and reinstalled Microsoft Authenticator on my new phone, but this does not fix the issue.
I installed and tried with Google Authenticator on my new phone and... tadaaaa... it works.
